sdk/eventhub/event-hubs/src/eventHubReceiver.ts

```diff
@@ -640,7 +645,7 @@ export class EventHubReceiver extends LinkEntity {
     if (this.receiverRuntimeMetricEnabled) {
       rcvrOptions.desired_capabilities = Constants.enableReceiverRuntimeMetricName;
     }
-    const eventPosition = options.eventPosition || this.options.beginReceivingAt;
+    const eventPosition = options.eventPosition || this.eventPosition;
```

Comment:
Why would `options` continue to have `eventPosition` here?

Reply:
options might have eventPosition set if the receiver has a `checkpoint` > -1. For example, when calling `detached`, the checkpoint is checked and the eventPosition is set.

Reply:
Then we will need a new type won't we?
Because `EventReceiverOptions` is exposed to users and should not have `eventPosition` anymore

Reply:
The `options` here isn't `EventReceiverOptions`, it is `CreateReceiverOptions` which is an internal interface and still has `eventPosition`.

Reply:
Ah! sorry, I thought we were still somewhere near the constructor that takes EventReceiverOptions!